services/auth/src/gcp.js

```diff
@@ -0,0 +1,106 @@
+const builder = require('./api');
+const {google} = require('googleapis');
+
+builder.declare({
+  method: 'get',
+  route: '/gcp/credentials/:projectId/:serviceAccount',
+  name: 'gcpCredentials',
+  output: 'gcp-credentials-response.yml',
+  stability: 'stable',
+  scopes: 'auth:gcp:access-token:<projectId>/<serviceAccount>',
+  title: 'Get Temporary Read/Write GCP Credentials',
+  description: [
+    'Get temporary GCP credentials for the given serviceAccount.',
+    'projectId must always be the string "-", which means "use the same',
+    'projectId as the account the service is running at.',
+    '',
+    'The call adds the necessary policy if the serviceAccount doesn\'t have it.',
+    'The credentials are set to expire after an hour, but this behavior is',
+    'subject to change. Hence, you should always read the `expires` property',
+    'from the response, if you intend to maintain active credentials in your',
+    'application.',
+  ].join('\n'),
+}, async function(req, res) {
+  const serviceAccount = req.params.serviceAccount;
+  const projectId = req.params.projectId;
+
+  if (projectId !== '-') {
```

Comment:
If we allow `projectId` to be `-` our scopes are going to look pretty funny :smile:  

But I guess that actually makes sense for when we move things around and such!

Reply:
It means that we'll have to make breaking changes when we move things around, is all.  I can push a change to this PR to fix that if you'd like..